lib/http/response/index.js

```diff
@@ -23,7 +24,7 @@ module.exports = function onResponse(res) {
     if (statusCode >= 400) {
       res.destroy();
       var he = new error.HttpError(statusCode);
-      he.url = res.req && res.req.path;
+      he.url = res.req && url.format(res.req);
```

Comment:
So, this will definitely not work. `res.req` is an instance of `http.ClientRequest`. The URL at this point is not easily recoverable and it may be ill advised to try and build it from the request object. I will dig a bit more.